java/23c-jdbc-bool-plsql/src/main/java/com/oracle/dev/jdbc/JDBCStoredProcHqEmployee.java

```diff
@@ -0,0 +1,115 @@
+/*
+  Copyright (c) 2024 Oracle and/or its affiliates.
+
+  This software is dual-licensed to you under the Universal Permissive License
+  (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License
+  2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose
+  either license.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package com.oracle.dev.jdbc;
+
+import java.sql.CallableStatement;
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.concurrent.ThreadLocalRandom;
+
+import oracle.jdbc.OracleConnection;
+import oracle.jdbc.pool.OracleDataSource;
+
+public class JDBCStoredProcHqEmployee {
+
+	
+
+	private final static String DB_URL = "jdbc:oracle:thin:@<DB_HOST>:<DB_PORT>/<DB_NAME>";
+	private final static String DB_USER = "<DB_USER>";
+	private final static String DB_PASSWORD = "<DB_PASSWORD>";
+	private static OracleConnection con;
+	private static CallableStatement stmt;
+
+	public static void main(String[] args) {
+
+		System.out.println("--------------------");
+		System.out.println("Input parameters");
+		System.out.println("--------------------");
+		
+		int id = ThreadLocalRandom.current().nextInt();
+		System.out.println("ID: " + id);
+
+		String name = "Duke";
+		System.out.println("Name: " + name);
+
+		String role = "Mascott";
+		System.out.println("Role: " + role);
+
+//		boolean active = true;		
+//		System.out.println("Active: " + active);
+
+		try {
+
+			Properties info = new Properties();
+			info.put(OracleConnection.CONNECTION_PROPERTY_USER_NAME, DB_USER);
+			info.put(OracleConnection.CONNECTION_PROPERTY_PASSWORD, DB_PASSWORD);
+			info.put(OracleConnection.CONNECTION_PROPERTY_FAN_ENABLED, false);
+
+			// JDBC datasource
+			OracleDataSource ods = new OracleDataSource();
+			ods.setURL(DB_URL);
+			ods.setConnectionProperties(info);
+
+			// JDBC connection
+			con = (OracleConnection) ods.getConnection();
+
+			// CallableStatement
+			// https://docs.oracle.com/en/java/javase/19/docs/api/java.sql/java/sql/CallableStatement.html
+			stmt = con.prepareCall("{call INSERT_HQ_EMPLOYEE_PRC(?,?,?,?,?)}");
```

Comment:
There was a PLSQL BOOLEAN type before 23c. What's new in 23c is the BOOLEAN type in SQL. What's interesting here is that you're using a BOOLEAN table column type and using it as a parameter type in a PLSQL proc. You should explain this in a comment. Also, there are native APIs in Java to bind a Boolean. I think it would add value to also show these APIs in addition to using setString(x, "1") that lets the server convert the String "1" to the boolean "true".